packages/angular/ssr/node/src/errors.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * Attaches listeners to the Node.js process to capture and handle unhandled rejections and uncaught exceptions.
+ * Captured errors are logged to the console. This function logs errors to the console, preventing unhandled errors
+ * from crashing the server. It is particularly useful for Zoneless apps, ensuring error handling without relying on Zone.js.
+ *
+ * @developerPreview
+ */
+export function attachNodeGlobalErrorHandlers(): void {
+  // Ensure that the listeners are registered only once.
+  // Otherwise, multiple instances may be registered during edit/refresh.
+  const gThis: typeof globalThis & { ngAttachNodeGlobalErrorHandlersCalled?: boolean } = globalThis;
+  if (gThis.ngAttachNodeGlobalErrorHandlersCalled) {
+    return;
+  }
+
+  gThis.ngAttachNodeGlobalErrorHandlersCalled = true;
+
+  process
+    // eslint-disable-next-line no-console
+    .on('unhandledRejection', (error) => console.error('unhandledRejection', error))
+    // eslint-disable-next-line no-console
+    .on('uncaughtException', (error) => console.error('uncaughtException', error));
```

Comment:
It's worth noting that this leaves the process in an undefined state which _may_ cause resource leaks and other unexpected behavior. See also: https://nodejs.org/api/process.html#warning-using-uncaughtexception-correctly

Could this behavior be made optional and come with "buyer beware"?

Reply:
+1 - discussions with @alxhub and @pkozlowski-opensource before going on leave were around _not_ doing this and needing the framework to handle application errors instead. 

This handler makes a choice about errors coming from more than just the framework/application.

Reply:
I totally agree, I wasn’t aware that things change since we last talked.

I definitely agree that the FW should handle errors.

Reply:
Agree as well - in general I would favor letting a process crash and restart if an error escapes the individual request scope. At that point, you can't really guarantee it can safely continue processing requests.

I wonder if there's potential for some kind of graceful shutdown though, where an uncaught error doesn't terminate the process immediately, but stops accepting new requests and gives some timeout for existing ones to shut down.

Reply:
Happy to chat more about options here. Unfortunately, there is (or at least used to be) no easy way to stop accepting new requests (vs. accepting new connections). IIRC the best strategy was to check on the end of requests if there's other concurrent requests on the same connection and using that to separate connections to proxies/load balancers.

Reply:
Should Angular servers implement `/healthz`? :D

Reply:
It's a really interesting open question because it says a lot about what this server is. Actually curious about what (if anything) other backend-for-frontend-in-a-box solutions do for this. Wouldn't be surprised if the answer is "not much" since it doesn't matter for most apps if a couple of concurrent requests end up with preventable 500s when something goes wrong..?